src/Components/LessonTemplate.jsx

```diff
@@ -23,18 +24,85 @@ export default function LessonTemplate({
   const { id } = useParams();
   const lesson_file = require("../Lesson/" + id + ".json");
   const answer = require("../Images/" + id + "_answer.png");
+  const [solution, setSolution] = useState();
+  const [result, setResult] = useState("closed");
+  const [showAnswer, setShowAnswer] = useState(false);
+  function solutionHandler(value) {
+    setSolution(value);
+  }
+  const closeModal = () => setResult("closed");
+  function runHandler() {
+    const nodes = solution.nodes.map((node) => {
+      if (node.id === "begin" || node.id === "end") {
+        return node.id;
+      } else {
+        return node.id.substring(0, node.id.length - 2);
+      }
+    });
+    const edges = solution.edges.map((edge) => {
+      const source =
+        edge.source === "begin" || edge.source === "end"
+          ? edge.source
+          : edge.source.substring(0, edge.source.length - 2);
+      const target =
+        edge.target === "begin" || edge.target === "end"
+          ? edge.target
+          : edge.target.substring(0, edge.target.length - 2);
+      return {
+        source: source,
+        target: target,
+        option: edge.data.selectedOption,
+      };
+    });
 
+    //compare solution with answer
+    const user_nodes = nodes.slice().sort();
+    const actual_nodes = lesson_file.solutions.nodes.slice().sort();
+    console.log(JSON.stringify(user_nodes) === JSON.stringify(actual_nodes));
+
+    const user_edges = edges.slice().sort((a, b) => {
+      const stringA = JSON.stringify(a);
+      const stringB = JSON.stringify(b);
+      if (stringA < stringB) return -1;
+      if (stringA > stringB) return 1;
+      return 0;
+    });
+    const actual_edges = lesson_file.solutions.edges.slice().sort((a, b) => {
+      const stringA = JSON.stringify(a);
+      const stringB = JSON.stringify(b);
+      if (stringA < stringB) return -1;
+      if (stringA > stringB) return 1;
+      return 0;
+    });
+    console.log(JSON.stringify(user_edges) === JSON.stringify(actual_edges));
```

Comment:
Plz clean up print statements

Reply:
got it

Reply:
@weikunwu i have remove console log and I will disable the background scroll next time.